File: frontend/src/app/api/models/ad-order-dto.ts
export interface AdOrderDto {
  id?: number,
  name: string,
  type: AdType,
  status: Status,
  onlyDesign: boolean,
  costCents: number,
  designs: string[],
}

export enum AdType {
  BILLBOARD = "BILLBOARD",
  BROADCAST = "BROADCAST",
  TRANSPORT = "TRANSPORT",
}

export enum Status {
  CREATED = "CREATED",
  UNDER_REVIEW = "UNDER_REVIEW",
  AWAITING_PAYMENT = "AWAITING_PAYMENT",
  DESIGNING = "DESIGNING",
  AWAITING_CONFIRMATION = "AWAITING_CONFIRMATION",
  COMPLETED = "COMPLETED",
  EXECUTING = "EXECUTING",
  IN_ACTION = "IN_ACTION",
  DISCONTINUED = "DISCONTINUED",
  CANCELED = "CANCELED",
}

File: backend/src/main/kotlin/com/vladrip/ifadvert/dto/AdOrderDto.kt
package com.vladrip.ifadvert.dto

import com.vladrip.ifadvert.entity.AdOrder

data class AdOrderDto (
    val id: Long?,
    val name: String,
    val type: AdOrder.AdType,
    val status: AdOrder.Status,
    val onlyDesign: Boolean,
    val costCents: Int,
    val designs: List<String>,
    val placements: List<AdOrder.Placement>,
)